packages/core/src/hooks/outputFiles.ts

```diff
@@ -5,19 +5,16 @@
 import path from 'path';
 
 import { HooksContext } from '../types';
-import { BuildPlugin } from '../webpack';
 import { formatDuration, writeFile } from '../helpers';
+import { BaseClass } from '../BaseClass';
 
 type Files = 'timings' | 'dependencies' | 'bundler' | 'metrics';
 
 type FilesToWrite = {
     [key in Files]?: { content: any };
 };
 
-const output = async function output(
-    this: BuildPlugin,
-    { report, metrics, bundler }: HooksContext
-) {
+const output = async function output(this: BaseClass, { report, metrics, bundler }: HooksContext) {
```

Comment:
Note: no need for `const output =`, you're already defining a function named `output`
```suggestion
async function output(this: BaseClass, { report, metrics, bundler }: HooksContext) {
```